rex/renewable_resource.py

```diff
@@ -66,8 +69,8 @@ def _preload_SAM(self, project_points, clearsky=False):
             Instance of SAMResource pre-loaded with Solar resource for sites
             in project_points
         """
-        SAM_res = SAMResource(project_points, self.time_index)
-        sites_slice = project_points.sites_as_slice
+        SAM_res = SAMResource(sites, tech, self.time_index)
+        sites_slice = SAM_res.sites_as_slice
```

Comment:
What happens if a slice can't be parsed? sites_slice gets assigned as a list right? Can we confirm we have a test for this?

Reply:
Follow up note, maybe the var shouldn't be called sites_slice if it can possibly be a list?

Reply:
this is your code, I just moved it from ProjectPoints, but agree on both!

Reply:
I'm not trying to play the blame game! Want to do edits or should I?

Reply:
I was just giving you a hard time, I completely agree. I just love it when we look back and go, WTF were we thinking there...

Reply:
okay renamed sites_as_slice to sites_slice and update the docstring

Reply:
Definitely. Haha I kind of meant that it should be like: `sites = SAM_res.sites_slice`. Thoughts?

Reply:
Thats exactly what I did, great minds!

Reply:
Working on tests now

Reply:
lol wait no, the variable name on the left of =. This is nitpicky but change it if you want.

Reply:
oh I see, I can do taht

Reply:
done and added a whole test_sam_resource file